### DISCUSSION TITLE
JP-extra分支的japanese_mora_list.py文件是不是有问题？

### DISCUSSION THREAD

**Discussion by uuKhost** [fishaudio/Bert-VITS2#349] [Q&A] [answered]
运行preprocess_text.py后，报错
![image](https://github.com/fishaudio/Bert-VITS2/assets/115860525/6b8fcca5-b866-4666-917b-3a6860106e91)
导致无法生成trainlist和vallist

**Comment by yin-zhang** [comment, marked as answer]
应该是python版本不够新导致的。前面加上
`from typing import Optional, List, Dict, Tuple`
然后把tuple, list, dict都换成Tuple, List, Dict就好了

**Reply by uuKhost** [reply]
确实是python版本的问题，之前别的分支3.8都能跑就没考虑这个，感谢回答
